File: tailwind-admin-reactjs-free/package/src/types/administrativo.ts
import type { Usuario } from 'src/types/estudiante';

export interface Administrativo {
  id_administrativo: number;
  name_administrativo: string;
  sec_name_administrativo: string;
  id_tipo_documento: number;
  num_documento_administrativo: string;
  phone_num_administrativo: string | null;
  direction_administrativo: string | null;
  cargo: string;
  usuario: Usuario;
}

export interface AdministrativoCreate {
  correo_institucional: string;
  password: string;
  name_administrativo: string;
  sec_name_administrativo: string;
  id_tipo_documento: number;
  num_documento_administrativo: string;
  phone_num_administrativo?: string | null;
  direction_administrativo?: string | null;
  cargo: string;
}

export interface AdministrativoUpdate {
  name_administrativo: string;
  sec_name_administrativo: string;
  id_tipo_documento: number;
  num_documento_administrativo: string;
  phone_num_administrativo?: string | null;
  direction_administrativo?: string | null;
  cargo: string;
}

File: backend/app/schemas/administrativo.py
from pydantic import BaseModel, ConfigDict, EmailStr, Field

from app.schemas.usuario import UsuarioOut


class AdministrativoBase(BaseModel):
    name_administrativo: str = Field(min_length=1, max_length=100)
    sec_name_administrativo: str = Field(min_length=1, max_length=100)
    id_tipo_documento: int
    num_documento_administrativo: str = Field(min_length=1, max_length=30)
    phone_num_administrativo: str | None = Field(default=None, max_length=20)
    direction_administrativo: str | None = Field(default=None, max_length=255)
    cargo: str = Field(min_length=1, max_length=100)


class AdministrativoCreate(AdministrativoBase):
    correo_institucional: EmailStr
    password: str = Field(min_length=8)


class AdministrativoUpdate(BaseModel):
    name_administrativo: str | None = Field(default=None, min_length=1, max_length=100)
    sec_name_administrativo: str | None = Field(default=None, min_length=1, max_length=100)
    id_tipo_documento: int | None = None
    num_documento_administrativo: str | None = Field(default=None, min_length=1, max_length=30)
    phone_num_administrativo: str | None = Field(default=None, max_length=20)
    direction_administrativo: str | None = Field(default=None, max_length=255)
    cargo: str | None = Field(default=None, min_length=1, max_length=100)


class AdministrativoOut(AdministrativoBase):
    model_config = ConfigDict(from_attributes=True)

    id_administrativo: int
    usuario: UsuarioOut
